src/search.cpp

```diff
@@ -365,24 +365,27 @@ void realMain(LambdaOptions     const & options)
 
         while (true)
         {
-            localHolder.reset();
-
-            // load records until batch is full or file at end
-            for (auto & [ id, seq ] : file_view)
+            if (localHolder.iterativeSearch != IterativeSearchMode::PHASE2)
             {
-                localHolder.qryIds.push_back(std::move(id));
-                localHolder.qrySeqs.push_back(std::move(seq));
+                localHolder.reset();
 
-                if (++localHolder.queryCount == globalHolder.records_per_batch)
-                    break;
-            }
+                // load records until batch is full or file at end
+                for (auto & [ id, seq ] : file_view)
+                {
+                    localHolder.qryIds.push_back(std::move(id));
+                    localHolder.qrySeqs.push_back(std::move(seq));
+
+                    if (++localHolder.queryCount == globalHolder.records_per_batch)
+                        break;
+                }
 
-            if (localHolder.queryCount == 0) // no more records in file
-                break;
+                if (localHolder.queryCount == 0) // no more records in file
+                    break;
 
-            globalHolder.queryCount += localHolder.queryCount; // atomic can be written from multiple threads
+                globalHolder.queryCount += localHolder.queryCount; // atomic can be written from multiple threads
+            }
 
-            localHolder.resetViews(); // views reset after sequences have been loaders
+            localHolder.resetViews(); // views reset after sequences have been loaded
```

Comment:
Why are the views resetted when in Phase 2?

Reply:
Because some of the query sequences (the successful ones) have been removed from the set and the views still refer to the old entries.

Reply:
(resetting views is a constant time operation, so it also doesn't matter in case no sequences were removed at all)